application/classes/Ushahidi/Formatter/Post/CSV.php

```diff
@@ -10,19 +10,30 @@
  */
 
 use Ushahidi\Core\SearchData;
-use Ushahidi\Core\Tool\Formatter;
 
-class Ushahidi_Formatter_Post_CSV implements Formatter
+use Ushahidi\Core\Tool\Filesystem;
+use Ushahidi\Core\Tool\FileData;
+use League\Flysystem\Util\MimeType;
+
+class Ushahidi_Formatter_Post_CSV extends Ushahidi_Formatter_API
 {
 	/**
 	 * @var SearchData
 	 */
 	protected $search;
+	protected $fs;
+	protected $tmpfname;
 
 	// Formatter
 	public function __invoke($records)
 	{
-		$this->generateCSVRecords($records);
+		return $this->generateCSVRecords($records);
+	}
+
+	public function setFS($fs)
```

Comment:
Total nitpick question: can we also call this setFilesystem() to parallel the other methods? And/or name it something to clarify that we're also setting a tempfname in here, if we don't want to move that into writeStreamToFS()?

Reply:
yes, my bad!
fixed